bin/generate-zbm

```diff
@@ -115,24 +121,44 @@ if ( ( defined $config{Global}{BootMountPoint} ) and ( length $config{Global}{Bo
 my $dir     = File::Temp->newdir();
 my $tempdir = $dir->dirname;
 
-# Set our kernel from the CLI, or pick the latest available in /boot
-if ( ( defined $runConf{kernel} ) and ( length $runConf{kernel} ) ) {
-  unless ( -f $runConf{kernel} ) {
-    printf "The provided kernel %s was not found, unable to continue", $runConf{kernel};
+if ( ! nonempty $runConf{kernel} ) {
+  # Try to determine a kernel file when one was not provided
+  if ( ! nonempty $runConf{kernel_version} ) {
+    $runConf{kernel} = latestKernel;
+  } else {
+    $runConf{kernel} = versionedKernel $runConf{kernel_version};
+  }
+
+  # Make sure a kernel was found
+  unless ( nonempty $runConf{kernel} ) {
+    print "Unable to choose a kernel file, cannot continue\n";
     exit;
   }
 } else {
-  $runConf{kernel} = latestKernel;
+  # Make sure the provided kernel file exists
+  unless ( -f $runConf{kernel} ) {
+    printf "The provided kernel %s was not found, unable to continue\n", $runConf{kernel};
+    exit;
+  }
 }
 
-$runConf{bootdir} = dirname( $runConf{kernel} );
-
-if ( basename( $runConf{kernel} ) =~ m/(vmlinux|vmlinuz|linux|kernel)-(\S+)/i ) {
-  $runConf{kernel_prefix} = $1;
-  $runConf{kernel_version} = $2;
-} else {
-  printf "Unable to determine kernel prefix and version from %s", $runConf{kernel};
-  exit;
+# Try to determine kernel_prefix or kernel_version if necessary
+if ( ! ( nonempty $runConf{kernel_prefix} and nonempty $runConf{kernel_version} ) ) {
```

Comment:
I wouldn't mind seeing `if ( ! ( ...` normalized to `unless ( ...`, just so the pattern is consistent throughout the whole tool.

Reply:
That's reasonable, done.